airflow/lineage/hook.py

```diff
@@ -159,6 +160,9 @@ def add_input_asset(
         asset_extra: dict | None = None,
     ):
         """Add the input asset and its corresponding hook execution context to the collector."""
+        if len(self._inputs) >= 100:
+            self.log.debug("Maximum number of asset inputs exceeded. Skipping.")
```

Comment:
Are you sure it's just debug? I guess this will be rather surprising for someone to see limited output without indication of what's happened?